narwhal/consensus/src/consensus.rs

```diff
@@ -337,3 +337,102 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::ConsensusState;
+    use fastcrypto::{bls12381::min_sig::BLS12381KeyPair, traits::KeyPair};
+    use rand::thread_rng;
+    use types::Certificate;
+
+    #[test]
+    pub fn test_gc() {
```

Comment:
nit: test functions don't need to be pub; other than that, we could use a bit more meaningful names to indicate what exactly is being tested wrt the GC